### PR TITLE
Data Stream support for Audit- Log

### PR DESCRIPTION
**Pull request by tmanninger**
### Description
Documentation for the MR https://github.com/opensearch-project/security/pull/4257

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
@tmanninger: Is there a doc issue associated with this PR?

File: _security/audit-logs/storage-types.md [hunk 1]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
 ```
-plugins.security.audit.type: <debug|internal_opensearch|external_opensearch|webhook|log4j>
+plugins.security.audit.type: <debug|internal_opensearch|internal_opensearch_data_stream|external_opensearch|webhook|log4j>
 ```
 
-`external_opensearch`, `webhook`, and `log4j` all have additional configuration options. Details follow.
+`internal_opensearch_data_stream`, `external_opensearch`, `webhook`, and `log4j` all have additional configuration options. Details follow.
+
+
+## Internal Opensearch Data Stream
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:33]:
```suggestion
## Internal OpenSearch data streams
```

File: _security/audit-logs/storage-types.md [hunk 2]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+
+The type `internal_opensearch_data_stream` have some additional configuration paramters.
+
+### Basics settings
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:37]:
```suggestion
### Basic settings
```

File: _security/audit-logs/storage-types.md [hunk 3]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+plugins.security.audit.type: <debug|internal_opensearch|internal_opensearch_data_stream|external_opensearch|webhook|log4j>
 ```
 
-`external_opensearch`, `webhook`, and `log4j` all have additional configuration options. Details follow.
+`internal_opensearch_data_stream`, `external_opensearch`, `webhook`, and `log4j` all have additional configuration options. Details follow.
+
+
+## Internal Opensearch Data Stream
+
+The type `internal_opensearch_data_stream` have some additional configuration paramters.
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:35]:
```suggestion
You can configure the `internal_opensearch_data_stream` type with the following configuration parameters.
```

File: _security/audit-logs/storage-types.md [hunk 4]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+
+### Basics settings
+
+Name | Data type | Description
+:--- | :--- | :---
+`plugins.security.audit.config.data_stream.name` | String | Name of the auditlog Data Stream. The Default is `opensearch-security-auditlog`.
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:41]:
```suggestion
`plugins.security.audit.config.data_stream.name` | String | The name of the audit log data stream. Default is `opensearch-security-auditlog`.
```

File: _security/audit-logs/storage-types.md [hunk 5]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+
+### Template settings
+
+Name | Data type | Description
+:--- | :--- | :---
+`plugins.security.audit.config.data_stream.template.manage` | Boolean | If enabled, the template for the Data Stream is managed by Opensearch. The Default is `true`.
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:47]:
```suggestion
`plugins.security.audit.config.data_stream.template.manage` | Boolean | If `true`, the template for the data stream is managed by Opensearch. Default is `true`.
```

File: _security/audit-logs/storage-types.md [hunk 6]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+`plugins.security.audit.config.data_stream.template.name` | String | Name of the template. The Default is `opensearch-security-auditlog`.
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:48]:
```suggestion
`plugins.security.audit.config.data_stream.template.name` | String | The name of the data stream template. Default is `opensearch-security-auditlog`.
```

File: _security/audit-logs/storage-types.md [hunk 7]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+`plugins.security.audit.config.data_stream.template.number_of_replicas` | Integer | Number of replicas for the Data Stream. The default is `0`.
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:49]:
```suggestion
`plugins.security.audit.config.data_stream.template.number_of_replicas` | Integer | The number of replicas for the data stream. Default is `0`.
```

File: _security/audit-logs/storage-types.md [hunk 8]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+`plugins.security.audit.config.data_stream.template.number_of_shards` | Integer | Number of Shards for the Data Stream. The default is `1`.
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:50]:
```suggestion
`plugins.security.audit.config.data_stream.template.number_of_shards` | Integer | The number of shards for the data stream. Default is `1`.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _security/audit-logs/storage-types.md [hunk 9]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
 
 You configure the output location in `opensearch.yml`:
 
 ```
-plugins.security.audit.type: <debug|internal_opensearch|external_opensearch|webhook|log4j>
+plugins.security.audit.type: <debug|internal_opensearch|internal_opensearch_data_stream|external_opensearch|webhook|log4j>
 ```
 
-`external_opensearch`, `webhook`, and `log4j` all have additional configuration options. Details follow.
+`internal_opensearch_data_stream`, `external_opensearch`, `webhook`, and `log4j` all have additional configuration options. Details follow.
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:30]:
```suggestion
`internal_opensearch_data_stream`, `external_opensearch`, `webhook`, and `log4j` can be customized with additional configuration options. For more information, see [Internal OpenSearch data streams](#internal-opensearch-data-streams).
```

File: _security/audit-logs/storage-types.md [hunk 10]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+
+
+## Internal OpenSearch data streams
+
+You can configure the `internal_opensearch_data_stream` type with the following configuration parameters.
+
+### Basic settings
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:37]: Will there be any additional settings called out in this section in the future?

File: _security/audit-logs/storage-types.md [hunk 11]
```diff
@@ -16,17 +16,38 @@ Setting | Description
...
+
+Name | Data type | Description
+:--- | :--- | :---
+`plugins.security.audit.config.data_stream.name` | String | The name of the audit log data stream. Default is `opensearch-security-auditlog`.
+
+### Template settings
+
+Name | Data type | Description
+:--- | :--- | :---
+`plugins.security.audit.config.data_stream.template.manage` | Boolean | If `true`, the template for the data stream is managed by Opensearch. Default is `true`.
```
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:47]:
```suggestion
`plugins.security.audit.config.data_stream.template.manage` | Boolean | When `true`, the template for the data stream is managed by Opensearch. Default is `true`.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _security/audit-logs/storage-types.md (hunk 10 above)
Comment (Naarcha-AWS) [inline _security/audit-logs/storage-types.md:37]:
```suggestion

```

**Comment by Naarcha-AWS** [comment]
@natebower: This is ready for editorial review.

**Review by Naarcha-AWS — APPROVED** [review]

File: _security/audit-logs/storage-types.md [hunk 12]
```diff
@@ -16,17 +16,37 @@ Setting | Description
...
+plugins.security.audit.type: <debug|internal_opensearch|internal_opensearch_data_stream|external_opensearch|webhook|log4j>
 ```
 
-`external_opensearch`, `webhook`, and `log4j` all have additional configuration options. Details follow.
+`internal_opensearch_data_stream`, `external_opensearch`, `webhook`, and `log4j` can be customized with additional configuration options. For more information, see [Internal OpenSearch data streams](#internal-opensearch-data-streams).
+
+
+## Internal OpenSearch data streams
+
+You can configure the `internal_opensearch_data_stream` type with the following configuration parameters.
```
Comment (natebower) [inline _security/audit-logs/storage-types.md:35]:
```suggestion
You can configure the `internal_opensearch_data_stream` type with the following parameters.
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS @tmanninger Just a minor deletion. Thanks!

**Pull request merged**